src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,208 @@
+    let i = from;
+    const range = [];
+
+    while (i <= to) {
+      range.push(i);
+      i += step;
+    }
+
+    return range;
+  };
+
+  /**
+   * Let's say we have 10 pages and we set pageNeighbours to 2
+   * Given that the current page is 6
+   * The pagination control will look like the following:
+   *
+   * (1) < {4 5} [6] {7 8} > (10)
+   *
+   * (x) => terminal pages: first and last page(always visible)
+   * [x] => represents current page
+   * {...x} => represents page neighbours
+   */
+  const fetchPageNumbers = () => {
+    /**
+     * totalNumbers: the total page numbers to show on the control
+     * totalBlocks: totalNumbers + 2 to cover for the left(<) and right(>) controls
+     */
+    const totalNumbers = pageNeighbors * 2 + 3;
+    const totalBlocks = totalNumbers + 2;
+
+    if (totalPages > totalBlocks) {
+      const startPage = Math.max(2, currentPage - pageNeighbors);
+      const endPage = Math.min(totalPages - 1, currentPage + pageNeighbors);
+
+      let pages = range(startPage, endPage);
+
+      /**
+       * hasLeftSpill: has hidden pages to the left
+       * hasRightSpill: has hidden pages to the right
+       * spillOffset: number of hidden pages either to the left or to the right
+       */
+      const hasLeftSpill = startPage > 2;
+      const hasRightSpill = totalPages - endPage > 1;
+      const spillOffset = totalNumbers - (pages.length + 1);
+
+      switch (true) {
+        // handle: (1) < {5 6} [7] {8 9} (10)
+        case hasLeftSpill && !hasRightSpill: {
+          const extraPages = range(startPage - spillOffset, startPage - 1);
+          pages = [LEFT_PAGE, ...extraPages, ...pages];
+          break;
+        }
+
+        // handle: (1) {2 3} [4] {5 6} > (10)
+        case !hasLeftSpill && hasRightSpill: {
+          const extraPages = range(endPage + 1, endPage + spillOffset);
+          pages = [...pages, ...extraPages, RIGHT_PAGE];
+          break;
+        }
+
+        // handle: (1) < {4 5} [6] {7 8} > (10)
+        case hasLeftSpill && hasRightSpill:
+        default: {
+          pages = [LEFT_PAGE, ...pages, RIGHT_PAGE];
+          break;
+        }
+      }
+
+      return [1, ...pages, totalPages];
+    }
+
+    return range(1, totalPages);
+  };
+
+  const gotoPage = page => {
+    const { onPageChanged = f => f } = props;
+
+    const currentPage = Math.max(0, Math.min(page, totalPages));
+
+    const paginationData = {
+      currentPage,
+      totalPages,
+      pageLimit,
+      totalRecords,
+    };
+
+    setCurrentPage(currentPage);
+    onPageChanged(paginationData);
+  };
+
+  const handleClick = page => {
+    gotoPage(page);
+  };
+
+  const handleMoveLeft = () => {
+    gotoPage(currentPage - pageNeighbors * 2 - 1);
+  };
+
+  const handleMoveRight = () => {
+    gotoPage(currentPage + pageNeighbors * 2 + 1);
+  };
+
+  return (
+    <Flex flexWrap="nowrap" justifyContent="center">
+      {pages.map((page, i) => {
+        return (
+          <PseudoBox
```

Comment:
@billy-le @racedale Big a11y issue here:

This should definitely not be a `<PseudoBox>` since those render plain old `<div>`'s which is a huge accessibility concern. This should be a `<Button>` since it has an action attached to it (not a link since it doesn't change the URL).